pkg/cache/v3/linear.go

```diff
@@ -320,34 +329,56 @@ func (cache *LinearCache) CreateWatch(request *Request, streamState stream.Strea
 	cache.mu.Lock()
 	defer cache.mu.Unlock()
 
+	// If the version is not up to date, check whether any requested resource has
+	// been updated between the last version and the current version. This avoids the problem
+	// of sending empty updates whenever an irrelevant resource changes.
+	stale := false
+	var staleResources map[string]struct{} // empty means all
+
 	if err != nil {
+		// The request does not include a version or the version could not be parsed.
+		// It will send all resources matching the request with no regards to the version.
```

Comment:
Avoid future tense in technical documentation. https://developers.google.com/style/tense

```suggestion
		// Send all resources matching the request with no regards to the version.
```

Reply:
Updated